### PR TITLE
fix: JSHandle preview text for non-ascii attributes/children

### PR DESCRIPTION
**Pull request by mxschmitt**
This surfaced in .NET that the string in the driver got incorrectly cut, then transferred to .NET as an invalid UTF8 character [`\ud835`](https://charbase.com/d835-unicode-invalid-character) which .NET wasn't able to parse and threw an error.

Drive-by: Move similar function from `packages/playwright-core/src/client/page.ts` into isomorphic `stringUtils`.

https://github.com/microsoft/playwright-dotnet/issues/2748

### CODE REVIEW

File: packages/playwright-core/src/utils/isomorphic/stringUtils.ts [hunk 1]
```diff
@@ -103,3 +103,10 @@ export function escapeForAttributeSelector(value: string | RegExp, exact: boolea
   // so we escape them differently.
   return `"${value.replace(/\\/g, '\\\\').replace(/["]/g, '\\"')}"${exact ? 's' : 'i'}`;
 }
+
+export function trimStringWithEllipsis(input: string, cap: number): string {
+  const chars = [...input];
```
Comment (dgozman) [inline packages/playwright-core/src/utils/isomorphic/stringUtils.ts:108]: To be more efficient, let's fast return:

```ts
if (input.length <= cap)
  return input;
```

File: packages/playwright-core/src/server/injected/injectedScript.ts [hunk 2]
```diff
@@ -1072,10 +1070,8 @@ export class InjectedScript {
       for (let i = 0; i < children.length; i++)
         onlyText = onlyText && children[i].nodeType === Node.TEXT_NODE;
     }
-    let text = onlyText ? (element.textContent || '') : (children.length ? '\u2026' : '');
-    if (text.length > 50)
-      text = text.substring(0, 49) + '\u2026';
```
Comment (dgozman) [inline packages/playwright-core/src/server/injected/injectedScript.ts:1077]: I wonder whether we have more `substring()` calls that produce bad UTF16? Can we somehow audit our usage?

Comment (mxschmitt) [reply packages/playwright-core/src/server/injected/injectedScript.ts:1077]: I did a quick audit by grepping for `.substring(0`, thats how I found the one in `page.ts`.

Comment (dgozman) [reply packages/playwright-core/src/server/injected/injectedScript.ts:1077]: I found this one as well: https://github.com/microsoft/playwright/blob/main/packages/playwright-core/src/server/injected/selectorGenerator.ts#L279

**Review by dgozman — APPROVED** [review]

**Pull request merged**